brilliant-builders/firestarter/game_engine/sprite.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Tuple
+
+from kivy.graphics import Rectangle
+from kivy.graphics.texture import Texture
+from kivy.uix.widget import Widget
+
+
+@dataclass
+class SpriteConfig:
+    file_path: Path
+    texture: Texture
+    size: List[int, int] = (64, 64)
+    animation_modes: int = 1
+    frame_count: List[int, ...] = (1,)
+    size_hint: List[int, int] = (64, 64)
```

Comment:
These type hints are immediately followed with an instantiation of a different type.